src/elements/content-sidebar/Sidebar.js

```diff
@@ -175,6 +182,46 @@ class Sidebar extends React.Component<Props, State> {
         }
     }
 
+    /**
+     * Handle keyboard interaction
+     *
+     * @param {Array<string>} tablist - Contains the pathnames of tabs
+     * @param {SyntheticKeyboardEvent} event - Keyboard event
+     */
+    onKeyDown = (tablist: Array<string>) => (event: SyntheticKeyboardEvent<>): void => {
+        const { history } = this.props;
+        const { nativeEvent } = event;
+
+        nativeEvent.stopImmediatePropagation();
+
+        const currIndex = tablist.indexOf(history.location.pathname);
+
+        switch (event.key) {
+            case KEYS.arrowUp:
+                if (currIndex === 0) {
+                    return;
+                }
+
+                history.replace(tablist[currIndex - 1]);
+
+                event.stopPropagation();
+                event.preventDefault();
+                break;
```

Comment:
history.replace(tablist[XYZ])
event.stopPropagation();
event.preventDefault();

can probably be made into a function

Reply:
the unit test is not complicated 😂